### PR TITLE
Add learning outcomes and knowledge checks for testing section

### LINKED ISSUE #22803: Add missing Knowledge Checks, Learning Outcomes & answer-links to JavaScript/Testing Basics
Related to #22472

This task is to go through the lesson referenced in the title of this issue and (as applicable):

- Add missing Learning Outcomes
- Add missing Knowledge Checks
- Add answer links in the Knowledge Checks

If the lesson already has Learning Outcomes or Knowledge Checks they do not need to be added to. Many lessons only need answer links added.

Knowledge check links should consist of an `<a>` tag which adds the `knowledge-check-link` class and links to the part of the lesson (or the external resource that is linked in the lesson) where the answer to the Knowledge Check can be found. Knowledge Check links should **not** point to external resources that weren't linked to as part of the lesson content or assignment. You can link to a specific line in the lesson like so:

```html
Lorem ipsum dolor sit amet, <span id="js-function-knowledge-check">consectetur adipiscing elit</span>, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to the page where the span with this id lives -->
<a class="knowledge-check-link" href="#js-function-knowledge-check">How do you return a value from a function?</a>
```
Or you can link to just a section of the lesson by just linking to the section heading and following [these](https://github.com/TheOdinProject/curriculum/wiki/Contributing-to-Wiki#anchor-linking) rules for determining how to format the link. Example:

```html
### Lorem ipsum dolor sit amet

sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to a heading on the page-->
<a class="knowledge-check-link" href="#lorem-ipsum-dolor-sit-amet">How do you return a value from a function?</a>
```

### Acceptance Criteria

- [ ] Lesson referenced in the title has Learning Outcomes, Knowledge Checks, and Knowledge Check links
- [ ] Knowledge Check links all link to the proper place in the lesson, or the proper external resource that was linked to earlier in the lesson
- [ ] Knowledge Check styling should **not** change

Please do not start work on this unless you've been assigned to it. Comment on this issue and ping me @wise-king-sullyman if you would like to be assigned to it.

Additionally, we would like for contributors to only take on one of these issues at a time. Please only request assignment for (and begin work on) an issue after the Learning Outcome/Knowledge Check/Knowledge Check link issue you're currently assigned to has been approved and merged.

Please check out [this new post on our wiki](https://github.com/TheOdinProject/curriculum/wiki/Learning-Outcomes-vs-Knowledge-Checks) about Learning Outcomes and Knowledge Checks if you need to write new Learning Outcomes or Knowledge Checks for this issue.

### LINKED ISSUE #22804: Add missing Knowledge Checks, Learning Outcomes & answer-links to JavaScript/More Testing
Related to #22472

This task is to go through the lesson referenced in the title of this issue and (as applicable):

- Add missing Learning Outcomes
- Add missing Knowledge Checks
- Add answer links in the Knowledge Checks

If the lesson already has Learning Outcomes or Knowledge Checks they do not need to be added to. Many lessons only need answer links added.

Knowledge check links should consist of an `<a>` tag which adds the `knowledge-check-link` class and links to the part of the lesson (or the external resource that is linked in the lesson) where the answer to the Knowledge Check can be found. Knowledge Check links should **not** point to external resources that weren't linked to as part of the lesson content or assignment. You can link to a specific line in the lesson like so:

```html
Lorem ipsum dolor sit amet, <span id="js-function-knowledge-check">consectetur adipiscing elit</span>, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to the page where the span with this id lives -->
<a class="knowledge-check-link" href="#js-function-knowledge-check">How do you return a value from a function?</a>
```
Or you can link to just a section of the lesson by just linking to the section heading and following [these](https://github.com/TheOdinProject/curriculum/wiki/Contributing-to-Wiki#anchor-linking) rules for determining how to format the link. Example:

```html
### Lorem ipsum dolor sit amet

sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to a heading on the page-->
<a class="knowledge-check-link" href="#lorem-ipsum-dolor-sit-amet">How do you return a value from a function?</a>
```

### Acceptance Criteria

- [ ] Lesson referenced in the title has Learning Outcomes, Knowledge Checks, and Knowledge Check links
- [ ] Knowledge Check links all link to the proper place in the lesson, or the proper external resource that was linked to earlier in the lesson
- [ ] Knowledge Check styling should **not** change

Please do not start work on this unless you've been assigned to it. Comment on this issue and ping me @wise-king-sullyman if you would like to be assigned to it.

Additionally, we would like for contributors to only take on one of these issues at a time. Please only request assignment for (and begin work on) an issue after the Learning Outcome/Knowledge Check/Knowledge Check link issue you're currently assigned to has been approved and merged.

Please check out [this new post on our wiki](https://github.com/TheOdinProject/curriculum/wiki/Learning-Outcomes-vs-Knowledge-Checks) about Learning Outcomes and Knowledge Checks if you need to write new Learning Outcomes or Knowledge Checks for this issue.

### PR DESCRIPTION
**Pull request by roshaan005**
- [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [x] The PR title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 
#### 1.Describe the changes made and include why they are necessary or important:
...your text here
For the testing javaScript section, knowledge checks and learning outcomes were added

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number below:
closes #22804 and closes #22803

### CODE REVIEW

**Comment by Maheshkumar-novice** [comment]
Hello, I think we should follow different format for the links. Now you are currently using format like this,
**-\[text](link)**
but we need to follow the below style
**-\<a class="knowledge-check-link" href="link">text</a>**

We need to use `class="knowledge-check-link"` to remove the default styles of the link. It's mentioned in the issue as well.

https://www.theodinproject.com/lessons/preview/ you can use this page to preview the links.

format
format1

**Comment by roshaan005** [comment]
Will fix asap , thank you for reviewing it!

File: javascript/testing/testing-1.md [hunk 1]
```diff
@@ -14,3 +25,10 @@ This lesson is going to center around __Jest__. The biggest reasons for this dec
...
 5. Watch [this amazing video](https://www.youtube.com/watch?v=URSWYvyc42M) that covers _what_ to test in your codebase.  The video is specifically about testing the Ruby language, but that doesn't matter _at all_.  The concepts here ring true in any language, and luckily Ruby is a clear enough language that you will be able to follow along just fine.  This video might be worth re-visiting after you've done some testing of your own.
 </div>
+
+
+### Knowledge Check
+- <a class = "knowledge-check-link" href = "#Introduction">What is TDD?</a>
+- <a class = "knowledge-check-link" href = "https://facebook.github.io/jest/docs/en/getting-started.html#content">How to set up and use jest?</a>
+- <a class = "knowledge-check-link" href = "http://godswillokwara.com/index.php/2016/09/09/the-importance-of-test-driven-development/">What are the benefits of writing unit tests?</a>
+- <a class = "knowledge-check-link" href = "https://dzone.com/articles/unit-testing-guidelines-what-to-test-and-what-not">What should you test in your codebase?</a>
```
Comment (twalton83) [inline javascript/testing/testing-1.md:34]: I may be missing it, but if we don’t link it in the lesson, we shouldn’t direct them to it in a Knowledge Check.

**Comment by wise-king-sullyman** [comment]
@roshaan005 if you're still working on these can you comment on issues #22803 and #22804 so that I can assign you, and update the opening comment from this PR with those issue numbers and the closing keyword (i.e. "closes #22803 if you're not familiar) so that when this is merged in the issues are closed?

File: javascript/testing/testing-2.md [hunk 2]
```diff
@@ -63,3 +72,10 @@ There are two solutions to the 'tightly coupled code' problem.  The first, and b
 4. Jest includes some _really_ handy mocking functions.  Read about them in the [official docs](https://facebook.github.io/jest/docs/en/mock-functions.html#content)
 5. And finally, if you wish, you can add Jest to your webpack setup.  Read about that process [here](https://facebook.github.io/jest/docs/en/webpack.html#content).
 </div>
+
+
+### Knowledge Checks
+- <a class = "knowledge-check-link" href = "#Pure-Functions">What are pure functions and how are they different from normal functions?</a>
```
[the remark this answers is not in the document]
Comment (twalton83) [reply javascript/testing/testing-2.md:78]: Do you mind updating this to fall in line with the knowledge checks formatting that can be found in other lessons?

**Review by twalton83 — CHANGES_REQUESTED** [review]

**Comment by twalton83** [comment]
@roshaan005 Are you still willing to make these changes? Otherwise I will close this tomorrow.

**Comment by twalton83** [comment]
Closing because stale.

**Pull request closed without merging**